### PR TITLE
Add more mappings for glibc stuff

### PR DESCRIPTION
**Pull request by alejandro-colomar**
Just another patch set adding mappings. :)

### CODE REVIEW

**Comment by alejandro-colomar** [comment]
Hi Kim,

Would you mind giving me some hints for fixing the tests? :)

Thanks,
Alex

**Comment by kimgr** [comment]
I'm pretty busy the coming two weeks and won't be able to get to this. Looks like the mappings have triggered an assertion, so you may want to follow that and see where it leads. If you have ideas for better debuggability, feel free to include them as separate commits or PRs.

**Comment by alejandro-colomar** [comment]
> I'm pretty busy the coming two weeks and won't be able to get to this.

Ah, don't worry; I can wait :)

> Looks like the mappings have triggered an assertion, so you may want to follow that and see where it leads. If you have ideas for better debuggability, feel free to include them as separate commits or PRs.

Well, I don't read pytest fluently.  It's not a matter of writing them better; it's the framework itself (and the reason is probably the Python language that isn't readable to me).  If it were me, tests would be written in sh(1); it would probably be hard to design a test suite from cratch with sh(1), but I guess tests would be very readable later.  But of course, I'm not asking you to rewrite the whole test suite from scratch at this point in time :p

**Comment by kimgr** [comment]
I think you can safely assume that the test framework works, do you shouldn't need to look at its source code. The tests are all driven by comments in testcases prefixed with `// IWYU.*`, they should give you enough hints on what is executed and expected to be able to troubleshoot.

**Comment by kimgr** [comment]
Just to be clear: the assertion failure comes from IWYU itself, not from the test framework. It appears to fail on conflicting mappings for `sys/syscall.h`:
```
.../iwyu_include_picker.cc:1870: Assertion failed: (*map)[key] == visibility Same file seen with two different visibilities: <sys/syscall.h> Old vis: 1 New vis: 2
```

And this stems from the existing mappings for `sys/syscall.h`:
* https://github.com/include-what-you-use/include-what-you-use/blob/master/iwyu_include_picker.cc#L454
* https://github.com/include-what-you-use/include-what-you-use/blob/master/iwyu_include_picker.cc#L532

which in turn are private because they ultimately map to `<syscall.h>`, the userspace-side header for syscalls.

I don't know offhand how to get your new mappings for syscall symbols in here, so I suggest you remove that commit to be able to test and reason about the other changes in isolation.

**Comment by kimgr** [comment]
Either way, let's handle that in a separate PR from the other, more straightforward (?), changes.

File: gcc.symbols.imp [hunk 1]
```diff
@@ -214,10 +217,634 @@
...
   { symbol: [ "FILE", private, "<stdio.h>", public ] },
+  { symbol: [ "MAP_POPULATE", private, "<sys/mman.h>", public ] },
+  { symbol: [ "MAP_POPULATE", private, "<linux/mman.h>", public ] },
   { symbol: [ "MAP_STACK", private, "<sys/mman.h>", public ] },
   { symbol: [ "MAP_STACK", private, "<linux/mman.h>", public ] },
+  { symbol: [ "MAXHOSTNAMELEN", private, "<sys/param.h>", public ] },
+  { symbol: [ "MAXHOSTNAMELEN", private, "<protocols/timed.h>", public ] },
   { symbol: [ "SIGCHLD", private, "<signal.h>", public ] },
   { symbol: [ "SIGCHLD", private, "<linux/signal.h>", public ] },
+  { symbol: [ "SYS_accept", private, "<sys/syscall.h>" public ] },
```
[the remark this answers is not in the document]
Comment (alejandro-colomar) [reply gcc.symbols.imp:228]: Heh!  And I proofread that line several times to check the script was correct.  xD

I'll fix it in a moment.  Luckily I pasted the script into the commit msg.

**Comment by alejandro-colomar** [comment]
> One thing that _might_ work is to turn it around and mark `<syscall.h>` as private and map it to a public `<sys/syscall.h>`, which now appears to be the go-to header.

That makes sense.  We wouldn't need to add the hundreds of mappings, right? :)

> Not sure about the status of `<syscall.h>` -- [this stackoverflow answer](https://stackoverflow.com/questions/7380591/what-happened-to-syscalls-h) indicates it's an undocumented convenience. shrug

I guess `<syscall.h>` was some supposed convenience a long time ago, but it's been never documented as far as I know, and it's non-portable, because some systems lack it, so the only portable and documented header is `<sys/syscall.h>`.
It's funny that the "convenience" is causing more problems than it's adding :P

**Comment by kimgr** [comment]
This looks good to me, let's get it merged.

**Pull request merged**